src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -317,5 +317,91 @@ private string FormatEnumValueToString(string value, JavaScriptEncoder? encoder)
 
             return converted;
         }
+
+        internal override T ReadWithQuotes(ref Utf8JsonReader reader)
+        {
+            string? enumString = reader.GetString();
+            Debug.Assert(enumString != null);
+
+            if (!Enum.TryParse(enumString, out T value)
+                && !Enum.TryParse(enumString, ignoreCase: true, out value))
+            {
+                ThrowHelper.ThrowJsonException();
+            }
+
+            return value;
+        }
+
+        internal override void WriteWithQuotes(Utf8JsonWriter writer, T value, JsonSerializerOptions options, ref WriteStack state)
```

Comment:
I don't think we can do that so easily since the differences between both methods are in the "end-calls", i.e. `Write` calls `WriteStringValue` and `WriteNumberValue`; `WriteWithQuotes` only calls `WritePropertyName`. if we move the logic to a helper method we would have to parameterize to know what's the right writer method to call.